pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -381,26 +381,54 @@ func (l *LibvirtDomainManager) preStartHook(vmi *v1.VirtualMachineInstance, doma
 	return domain, err
 }
 
-// This function parses the SRIOV-VF-PCI-ADDR variable that is set by SR-IOV
-// device plugin listing PCI IDs for devices allocated to the pod. The format
-// is as follows:
+// This function parses variables that are set by SR-IOV device plugin listing
+// PCI IDs for devices allocated to the pod. It also parses variables that
+// virt-controller sets mapping network names to their respective resource
+// names (if any).
 //
+// Format for PCI ID variables set by SR-IOV DP is:
 // "": for no allocated devices
-// "0000:81:11.1,": for a single device
-// "0000:81:11.1,0000:81:11.2[,...]": for multiple devices
-func getSRIOVPCIAddresses() []string {
-	pciAddrString, isSet := os.LookupEnv("SRIOV-VF-PCI-ADDR")
-	if isSet {
-		addrs := strings.Split(pciAddrString, ",")
-		naddrs := len(addrs)
-		if naddrs > 0 {
-			if addrs[naddrs-1] == "" {
-				addrs = addrs[:naddrs-1]
+// PCIDEVICE_<resourceName>="0000:81:11.1": for a single device
+// PCIDEVICE_<resourceName>="0000:81:11.1 0000:81:11.2[ ...]": for multiple devices
+//
+// Since special characters in environment variable names are not allowed,
+// resourceName is mutated as follows:
+// 1. All dots and slashes are replaced with underscore characters.
+// 2. The result is upper cased.
+//
+// Example: PCIDEVICE_INTEL_COM_SRIOV_TEST=... for intel.com/sriov_test resources.
+//
+// Format for network to resource mapping variables is:
+// KUBEVIRT_RESOURCE_NAME_<networkName>=<resourceName>
+//
+func resourceNameToEnvvar(resourceName string) string {
+	varName := strings.ToUpper(resourceName)
+	varName = strings.Replace(varName, "/", "_", -1)
+	varName = strings.Replace(varName, ".", "_", -1)
+	return fmt.Sprintf("PCIDEVICE_%s", varName)
+}
+
+func getSRIOVPCIAddresses(ifaces []v1.Interface) map[string][]string {
+	networkToAddressesMap := map[string][]string{}
+	for _, iface := range ifaces {
+		networkToAddressesMap[iface.Name] = []string{}
+		varName := fmt.Sprintf("KUBEVIRT_RESOURCE_NAME_%s", iface.Name)
+		resourceName, isSet := os.LookupEnv(varName)
+		if isSet {
+			pciAddrString, isSet := os.LookupEnv(resourceNameToEnvvar(resourceName))
+			if isSet {
+				addrs := strings.Split(pciAddrString, ",")
+				naddrs := len(addrs)
+				if naddrs > 0 {
+					if addrs[naddrs-1] == "" {
+						addrs = addrs[:naddrs-1]
+					}
+				}
+				networkToAddressesMap[iface.Name] = addrs
```

Comment:
I think we should log warning if sr-iov network was requested but environment variable is not provided, for easier debugging.